lib/gollum-lib/filter/toc.rb

```diff
@@ -87,8 +86,11 @@ def generate_anchor_name(header)
   # Creates an anchor element with the given name and adds it before
   # the given header element.
   def add_anchor_to_header(header, name)
-    anchor_element = %Q(<a class="anchor" id="#{name}" href="##{name}"></a>)
-    header.children.before anchor_element # Add anchor element before the header
+    a = Nokogiri::XML::Node.new('a', @doc)
+    a['class'] = 'anchor'
+    a['id'] = name
+    a['href'] = "##{name}"
+    header.children.before a # Add anchor element before the header
```

Comment:
Maybe `header.children.before(a)` for consistency with parentheses? Otherwise this can be merged as far as I'm concerned. Nice find and good fix!